great_expectations/expectations/metrics/column_aggregate_metrics/column_distinct_values.py

```diff
@@ -30,9 +33,18 @@ def _sqlalchemy(
         metric_value_kwargs: Dict,
         metrics: Dict[str, Any],
         runtime_configuration: Dict,
-    ):
-        observed_value_counts = metrics["column.value_counts"]
-        return set(observed_value_counts.index)
+    ) -> Set[Any]:
+        selectable, _, accessor_domain_kwargs = execution_engine.get_compute_domain(
+            metric_domain_kwargs, MetricDomainTypes.COLUMN
+        )
+        column = accessor_domain_kwargs["column"]
+
+        query = sa.select(sa.column(column)).distinct()
+        results = execution_engine.engine.execute(
+            query.select_from(selectable)
+        ).fetchall()
+
+        return {row[0] for row in results}
```

Comment:
Vectorized operation is not faster here due to overhead of converting to and from numpy array.

Reply:
Maybe add this as a comment in the code since someone may look at this and spend cycles doing that.

Reply:
@NathanFarmer If this new implementation is not faster, should we keep the old one (because it requires fewer lines, due to reusing the existing machinery)?

Reply:
@alexsherstinsky `column.value_counts` should not be required for `column.distinct_values`. 1) `column.value_counts` is a slower query that we should try to avoid if possible and 2) If possible we should not be passing around in-memory data with extra columns (counts) that we don't need.